### PR TITLE
Update README.md 

### PR DESCRIPTION
**Pull request by SimonDallaway**
Update the README in response to https://issues.jboss.org/browse/DROOLS-4399

### CODE REVIEW

**Comment by kie-ci2** [comment]
Can one of the admins verify this PR? Comment with 'ok to test' to start the build.

File: kie-server-parent/kie-server-wars/kie-server/src/main/webc-resources/README.md [hunk 1]
```diff
@@ -6,29 +6,33 @@ This instruction describes all steps to install KIE Server on Tomcat 9 standalon
...
-   - org.slf4j:artifactId=slf4j-jdk14
+   - org.slf4j:slf4j-api
+   - org.slf4j:slf4j-jdk14
    - org.jboss.spec.javax.transaction:jboss-transaction-api_1.2_spec
    - org.jboss.integration:narayana-tomcat
    - org.jboss.narayana.jta:narayana-jta
    - org.jboss:jboss-transaction-spi
+   - org.jboss.logging:jboss-logging
 
  versions of these libraries will depend on the release, so best to check what versions are shipped with KIE
```
Comment (mswiderski) [inline kie-server-parent/kie-server-wars/kie-server/src/main/webc-resources/README.md]: Here are all the versions defined https://github.com/kiegroup/droolsjbpm-build-bootstrap/blob/master/pom.xml#L56

File: kie-server-parent/kie-server-wars/kie-server/src/main/webc-resources/README.md [hunk 2]
```diff
@@ -38,38 +42,29 @@ This instruction describes all steps to install KIE Server on Tomcat 9 standalon
...
+        factory="org.apache.tomcat.jdbc.naming.GenericNamingResourcesFactory"
+		loginTimeout="0" 
         name="xads"
         uniqueName="xads" 
+        testOnBorrow="false" 
+        driverType="4"
+
         portNumber="${datasource.port}"
         serverName="${datasource.hostname}" 
-        testOnBorrow="false"
```
Comment (mswiderski) [inline kie-server-parent/kie-server-wars/kie-server/src/main/webc-resources/README.md]: Why is this removed? Did it cause any issues ?

Comment (rsynek) [reply kie-server-parent/kie-server-wars/kie-server/src/main/webc-resources/README.md]: it's not removed, just the order of attributes has been changed.

Comment (rsynek) [reply kie-server-parent/kie-server-wars/kie-server/src/main/webc-resources/README.md]: @SimonDallaway was the order of attributes changed to comply with some commonly used convention?

File: kie-server-parent/kie-server-wars/kie-server/src/main/webc-resources/README.md [hunk 3]
```diff
@@ -38,38 +42,29 @@ This instruction describes all steps to install KIE Server on Tomcat 9 standalon
     As a next step, configure a pooling data source, that relies on XA data source for creating new connections. 
     In the example, this data source is named “poolingXaDs”.
     
-    Edit TOMCAT_HOME/conf/context.xml and add following within Context tags of the file:
-    ``` 
-    <Resource 
+    Edit TOMCAT_HOME/conf/server.xml and add following within GlobalNamingResources tags of the file (e.g. after the UserDatabase Resource definition):
```
Comment (rsynek) [inline kie-server-parent/kie-server-wars/kie-server/src/main/webc-resources/README.md]: @mswiderski please note that maven cargo creates all the declared datasource resources [1] in context.xml. Thus if we change it here, we don't test the exactly same configuration any more.

[1] https://github.com/kiegroup/droolsjbpm-integration/blob/master/kie-server-parent/kie-server-tests/pom.xml#L629

File: kie-server-parent/kie-server-wars/kie-server/src/main/webc-resources/README.md (hunk 2 above)
Comment (SimonDallaway) [reply kie-server-parent/kie-server-wars/kie-server/src/main/webc-resources/README.md]: Hi,

I think I was just trying to ensure that the properties with literal values were grouped first followed by the ones that were variables.

My major concern was getting the instructions so that they worked and were presented logically. I'm a definite newbie to all of this.

File: kie-server-parent/kie-server-wars/kie-server/src/main/webc-resources/README.md (hunk 1 above)
Comment (SimonDallaway) [reply kie-server-parent/kie-server-wars/kie-server/src/main/webc-resources/README.md]: @mswiderski do you want me to make the update in my fork?

Comment (mswiderski) [reply kie-server-parent/kie-server-wars/kie-server/src/main/webc-resources/README.md]: yes, please go ahead

File: kie-server-parent/kie-server-wars/kie-server/src/main/webc-resources/README.md (hunk 3 above)
Comment (mswiderski) [reply kie-server-parent/kie-server-wars/kie-server/src/main/webc-resources/README.md]: @rsynek then the question is why that didn't work for @SimonDallaway?

Comment (rsynek) [reply kie-server-parent/kie-server-wars/kie-server/src/main/webc-resources/README.md]: The only difference I see is the usage of custom XARecovery configuration. I quickly glanced over the webc distribution and test setup and I don't see it being used anywhere.

**Comment by mareknovotny** [comment]
@rsynek what will we do with this PR?

File: kie-server-parent/kie-server-wars/kie-server/src/main/webc-resources/README.md [hunk 4]
```diff
@@ -114,26 +126,28 @@ This instruction describes all steps to install KIE Server on Tomcat 9 standalon
...
     
     - setenv.bat:
-    -Dcom.arjuna.ats.jta.recovery.XAResourceRecovery1=com.arjuna.ats.internal.jdbc.recovery.BasicXARecovery;abs://$CATALINA_HOME/conf/xa-recovery-properties.xml" ";1
-
+    -Dcom.arjuna.ats.jta.recovery.XAResourceRecovery1=com.arjuna.ats.internal.jdbc.recovery.BasicXARecovery;abs://$CATALINA_HOME/conf/xa-recovery-properties.xml"
 
     BasicXARecovery supports following parameters:
      - path to the properties file
-     - the number of connections defined in the properties file
+     - the number of connections defined in the properties file; this doesn't work until the fix for https://issues.jboss.org/browse/JBTM-3170 is incorporated into the release
```
Comment (gmunozfe) [inline kie-server-parent/kie-server-wars/kie-server/src/main/webc-resources/README.md:153]: Currently narayana version is 5.9, where this bug was already fixed, so this clarification is no longer needed:
```
mvn dependency:tree | grep narayana-jta
[INFO] +- org.jboss.narayana.jta:narayana-jta:jar:5.9.0.Final:test
```

Comment (mareknovotny) [reply kie-server-parent/kie-server-wars/kie-server/src/main/webc-resources/README.md:153]: ok, does it mean this setup was also verified by you @gmunozfe practically by tests or manually?
Because we talked with @rsynek that this probably still needs real verification

Comment (gmunozfe) [reply kie-server-parent/kie-server-wars/kie-server/src/main/webc-resources/README.md:153]: @mareknovotny, I've tested with current libraries and followed the discussion from in the jira above.

As @SimonDallaway stated, there's an exception at startup due to JNDI _NameNotFoundException_, that doesn't happen when moving the configuration to **server.xml** under _GlobalNamingResources_. 

`WARN [Periodic Recovery] sun.reflect.NativeMethodAccessorImpl.invoke0 ARJUNA016009: Caught:
	java.sql.SQLException: javax.naming.NameNotFoundException: Name [comp/env/xads] is not bound in this Context. Unable to find [comp].`

That made me think this change was necessary, however after a deeper analysis this initial exception doesn't avoid _[Periodic Recovery]_ thread to be executed and worked out:

` [Periodic Recovery] com.arjuna.ats.internal.jta.recovery.arjunacore.XARecoveryModule.xaRecoveryFirstPass xarecovery of com.mysql.jdbc.jdbc2.optional.JDBC4MysqlXAConnection@5a826a4b
 DEBUG [Periodic Recovery] com.arjuna.ats.internal.jta.recovery.arjunacore.XARecoveryModule.xaRecoveryFirstPass Found 2 xids in doubt`

So, except moving the configuration from context.xml to web.xml (we can keep it as it is, and that part can be added as an alternative configuration), I agree with the rest of PR, because it adds some clarifications useful to the community.

I've tested it with mysql
`type="com.mysql.jdbc.jdbc2.optional.MysqlXADataSource"
`

and also extra configuration was needed in the setenv.sh script (specific for mysql):
`
-Dorg.kie.server.
persistence.dialect=org.hibernate.dialect.MySQL5InnoDBDialect
`

Summarizing, @SimonDalaway, if you agree, perhaps you could rephrase it to keep both configurations, stating that setting the configuration at server.xml allows multiple applications running in the same Tomcat instance to share the database connection pool.

Comment (SimonDallaway) [reply kie-server-parent/kie-server-wars/kie-server/src/main/webc-resources/README.md]: Hi Gonzalo,

Tweaked it. Hopefully all good now.

Simon

Comment (MarianMacik) [reply kie-server-parent/kie-server-wars/kie-server/src/main/webc-resources/README.md]: @gmunozfe So to sum up, this exception was just an initial warning and after that it works as well?

Comment (gmunozfe) [reply kie-server-parent/kie-server-wars/kie-server/src/main/webc-resources/README.md]: Right @MarianMacik,
Exception is at first scan at [resourceInitiatedRecovery](https://github.com/jbosstm/narayana/blob/5.9.0.Final/ArjunaJTA/jta/classes/com/arjuna/ats/internal/jta/recovery/arjunacore/XARecoveryModule.java#L194), but later, there's an additional scan adding those from [resourceInitiatedRecoveryForRecoveryHelpers](https://github.com/jbosstm/narayana/blob/5.9.0.Final/ArjunaJTA/jta/classes/com/arjuna/ats/internal/jta/recovery/arjunacore/XARecoveryModule.java#L196) which retrieves correctly the resources (XAConnection) and therefore, works at [xaRecoveryFirstPass](https://github.com/jbosstm/narayana/blob/5.9.0.Final/ArjunaJTA/jta/classes/com/arjuna/ats/internal/jta/recovery/arjunacore/XARecoveryModule.java#L201)

File: kie-server-parent/kie-server-wars/kie-server/src/main/webc-resources/README.md [hunk 5]
```diff
@@ -114,25 +127,27 @@ This instruction describes all steps to install KIE Server on Tomcat 9 standalon
...
+        <entry key="DB_1_DatabaseURL">java:xads</entry>
     </properties>
+```
 
+    This causes the recovery process to utilise the underlying data source defined in the GlobalNamingResources section of the server.xml file.
     Append to CATALINA_OPTS in setenv.sh|bat file following:
     
     - setenv.sh:
-    -Dcom.arjuna.ats.jta.recovery.XAResourceRecovery1=com.arjuna.ats.internal.jdbc.recovery.BasicXARecovery\;abs://$CATALINA_HOME/conf/xa-recovery-properties.xml\ \;1
+    -Dcom.arjuna.ats.jta.recovery.XAResourceRecovery1=com.arjuna.ats.internal.jdbc.recovery.BasicXARecovery\;abs://$CATALINA_HOME/conf/xa-recovery-properties.xml
```
Comment (gmunozfe) [inline kie-server-parent/kie-server-wars/kie-server/src/main/webc-resources/README.md:147]: @SimonDallaway do you mind to undo it here and keep it as it was before?

File: kie-server-parent/kie-server-wars/kie-server/src/main/webc-resources/README.md [hunk 6]
```diff
@@ -114,25 +127,27 @@ This instruction describes all steps to install KIE Server on Tomcat 9 standalon
...
     
     - setenv.bat:
-    -Dcom.arjuna.ats.jta.recovery.XAResourceRecovery1=com.arjuna.ats.internal.jdbc.recovery.BasicXARecovery;abs://$CATALINA_HOME/conf/xa-recovery-properties.xml" ";1
-
+    -Dcom.arjuna.ats.jta.recovery.XAResourceRecovery1=com.arjuna.ats.internal.jdbc.recovery.BasicXARecovery;abs://$CATALINA_HOME/conf/xa-recovery-properties.xml"
```
Comment (gmunozfe) [inline kie-server-parent/kie-server-wars/kie-server/src/main/webc-resources/README.md:150]: @SimonDallaway same here

Comment (SimonDallaway) [reply kie-server-parent/kie-server-wars/kie-server/src/main/webc-resources/README.md]: Done

File: kie-server-parent/kie-server-wars/kie-server/src/main/webc-resources/README.md (hunk 4 above)
Comment (mareknovotny) [reply kie-server-parent/kie-server-wars/kie-server/src/main/webc-resources/README.md]: thanks guys for clarification and review

**Comment by mareknovotny** [comment]
if everyone agrees, I will merge it now
